### PR TITLE
Add improved VerifyNoOutstandingExpectation messages

### PR DESCRIPTION
**Pull request by hwoodiwiss**
Implementation for #78

### CODE REVIEW

**Comment by hwoodiwiss** [comment]
I've added tests

@richardszalay would you mind taking a look at this?

File: RichardSzalay.MockHttp.Shared/MockedRequestExtensions.cs [hunk 1]
```diff
@@ -14,16 +15,17 @@ namespace RichardSzalay.MockHttp
...
-            return source.With(new CustomMatcher(matcher));
+    {
+		/// <summary>
+		/// Constraints the request using custom logic
+		/// </summary>
+		/// <param name="source">The source mocked request</param>
+		/// <param name="matcher">The delegate that will be used to constrain the request</param>
+		/// <param name="matcherExpression">This will be automatically populated with the text of the matcher when using C# 10</param>
+		/// <returns>The <see cref="T:MockedRequest"/> instance</returns>
+		public static MockedRequest With(this MockedRequest source, Func<HttpRequestMessage, bool> matcher, [CallerArgumentExpression("matcher")] string matcherExpression = "")
```
Comment (richardszalay) [inline RichardSzalay.MockHttp.Shared/MockedRequestExtensions.cs:26]: Can we make this an overload? Otherwise it's breaking the public API.

**Review by richardszalay — CHANGES_REQUESTED** [review]
Thanks for submitting this PR @hwoodiwiss, and apologies for taking so long to reply.

I'll need to publish this as a major/breaking change release due to the change in the public interfaces, but I think it's worth it.

File: RichardSzalay.MockHttp.Shared/MockedRequestExtensions.cs (hunk 1 above)
Comment (hwoodiwiss) [reply RichardSzalay.MockHttp.Shared/MockedRequestExtensions.cs:26]: Yeah, happy to revert the change to With. 
Given that CallerArgumentExpression needs an empty string default parameter to be able to populate, I think we might need to add a new method (WithExpression?), as without, I don't think the compiler will ever choose the overload new overload.

**Comment by hwoodiwiss** [comment]
@richardszalay I've been thinking it over, it might be easiest for now to remove the CallerArgumentExpression stuff, and add that later as a separate feature, what do you think?

**Comment by hwoodiwiss** [comment]
Closing this as I've had not had a response from the maintainer in a couple of years (not complaining, people have lives). If anyone else wants to look at implementing something similar in the future, feel free to use/adapt any of the code I've written here as needed.

**Comment by hwoodiwiss** [comment]
No problem, I'll have to have a play with 7.0.0!

Thank you for the library and the time you give to the community!

**Comment by richardszalay** [comment]
Thanks Hugo, and thank you again for the PR. I don't like not merging PRs, but from memory the reason I went for a different implementation is that I decided to also solve for some other usecases.

**Pull request closed without merging**